crates/oxc_linter/src/rules/eslint/new_cap.rs

```diff
@@ -0,0 +1,800 @@
+        ),
+        ("var x = Foo(42);", Some(serde_json::json!([{ "capIsNewExceptionPattern": "^Foo" }]))),
+        (
+            "var x = new foo(42);",
+            Some(serde_json::json!([{ "newIsCap": true, "newIsCapExceptions": ["foo"] }])),
+        ),
+        ("var x = new foo(42);", Some(serde_json::json!([{ "newIsCapExceptionPattern": "^foo" }]))),
+        ("var x = Object(42);", Some(serde_json::json!([{ "capIsNewExceptions": ["Foo"] }]))),
+        ("var x = Foo.Bar(42);", Some(serde_json::json!([{ "capIsNewExceptions": ["Bar"] }]))),
+        ("var x = Foo.Bar(42);", Some(serde_json::json!([{ "capIsNewExceptions": ["Foo.Bar"] }]))),
+        (
+            "var x = Foo.Bar(42);",
+            Some(serde_json::json!([{ "capIsNewExceptionPattern": "^Foo\\.." }])),
+        ),
+        ("var x = new foo.bar(42);", Some(serde_json::json!([{ "newIsCapExceptions": ["bar"] }]))),
+        (
+            "var x = new foo.bar(42);",
+            Some(serde_json::json!([{ "newIsCapExceptions": ["foo.bar"] }])),
+        ),
+        (
+            "var x = new foo.bar(42);",
+            Some(serde_json::json!([{ "newIsCapExceptionPattern": "^foo\\.." }])),
+        ),
+        ("var x = new foo.bar(42);", Some(serde_json::json!([{ "properties": false }]))),
+        ("var x = Foo.bar(42);", Some(serde_json::json!([{ "properties": false }]))),
+        (
+            "var x = foo.Bar(42);",
+            Some(serde_json::json!([{ "capIsNew": false, "properties": false }])),
+        ),
+        ("foo?.bar();", None),       // { "ecmaVersion": 2020 },
+        ("(foo?.bar)();", None),     // { "ecmaVersion": 2020 },
+        ("new (foo?.Bar)();", None), // { "ecmaVersion": 2020 },
+        ("(foo?.Bar)();", Some(serde_json::json!([{ "properties": false }]))), // { "ecmaVersion": 2020 },
+        ("new (foo?.bar)();", Some(serde_json::json!([{ "properties": false }]))), // { "ecmaVersion": 2020 },
+        ("Date?.UTC();", None),   // { "ecmaVersion": 2020 },
+        ("(Date?.UTC)();", None), // { "ecmaVersion": 2020 }
+    ];
+
+    let fail = vec![
+        ("var x = new c();", None),
+        ("var x = new φ;", None),
+        ("var x = new a.b.c;", None),
+        ("var x = new a.b['c'];", None),
+        ("var b = Foo();", None),
+        ("var b = a.Foo();", None),
+        ("var b = a['Foo']();", None),
+        ("var b = a.Date.UTC();", None),
+        ("var b = UTC();", None),
+        ("var a = B.C();", None),
+        (
+            "var a = B
+			.C();",
+            None,
+        ),
+        ("var a = new B.c();", None),
+        (
+            "var a = new B.
+			c();",
+            None,
+        ),
+        ("var a = new c();", None),
+        ("var a = new b[ ( 'foo' ) ]();", None), // { "ecmaVersion": 6 },
+        ("var a = new b[`foo`];", None),         // { "ecmaVersion": 6 },
+        // (
+        //     "var a = b[`\\
+        // 	Foo`]();",
+        //     None,
+        // ), // { "ecmaVersion": 6 },
```

Comment:
I do not know why this should fail. we are ignoring non alpha character (`GetCapResult`)